netbox/ipam/forms/model_forms.py

```diff
@@ -21,7 +21,7 @@
 from utilities.forms.utils import get_field_value
```

Comment:
If you set parent type without the associated device, VM, FHRP Group and try to save there is no error given, should be a "Cannot set parent type without an associated object" message

Reply:
Phew, this only took forever to figure out. After a fair amount of debugging compiled .ts, reading bootstrap docs, and finally understanding how those in-page "This field is required." messages get displayed, I discovered this is because of #18916.

The **tl;dr** is that the `APISelector`'s template changes the DOM tree structure just enough to break the way Bootstrap display's invalid feedback for fields. See [my comment on the issue](https://github.com/netbox-community/netbox/issues/18916#issuecomment-2791189429).

Reply:
I think the work-around for this one is to just use Django's built in `Select` widget, since with the `HTMXSelect` widget on the `parent_object_type` form field, the form is being rendered and a queryset attached to the `parent` form field select, we don't necessarily need the `APISelect` widget functionality.

The downside is that means no object selector modal for the `parent` form field for the time being.

Alternatively, we solve #18916 real quick.

Reply:
After discussing this in the maintainers call, we decided to move forward with this work as is re: the lack of validation messaging working in the `DynamicModelChoiceField`. #18916 will be a follow up in a future patch release.